task.sql

```diff
@@ -4,30 +4,66 @@
 USE ShopDB;
 
 CREATE TABLE Countries (
-    ID INT,
+    ID INT AUTO_INCREMENT,
     Name VARCHAR(50),
     PRIMARY KEY (ID)
 );
 
+CREATE TABLE Cities (
+		ID INT AUTO_INCREMENT,
+		Name VARCHAR(50),
+		CountryID INT,
+		FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
+		PRIMARY KEY (ID)
+);
+
+CREATE TABLE Addresses (
+		ID INT AUTO_INCREMENT,
+		StreetName VARCHAR(100),
+		CityID INT,
+		FOREIGN KEY (CityID) REFERENCES Cities(ID) ON DELETE NO ACTION,
+		PRIMARY KEY (ID)
+);
+
+CREATE TABLE Warehouses (
+		ID INT AUTO_INCREMENT,
+		Name VARCHAR(50),
+		AddressID INT,
+		FOREIGN KEY (AddressID) REFERENCES Addresses(ID) ON DELETE NO ACTION,
+		PRIMARY KEY (ID)
+)
+
 CREATE TABLE ProductInventory (
-    ID INT,
-    ProductName VARCHAR(50),
-    WarehouseAmount INT,
-    WarehouseName VARCHAR(50),
-    WarehouseAddress VARCHAR(50), 
-    CountryID INT,
-	FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
-    PRIMARY KEY (ID)
+		ID INT AUTO_INCREMENT,
+		ProductName VARCHAR(50),
+		WarehouseAmount INT,
+		WarehouseID INT,
+		FOREIGN KEY (WarehouseID) REFERENCES Warehouses(ID) ON DELETE NO ACTION,
```

Comment:
The foreign key constraint `ON DELETE NO ACTION` for `WarehouseID` might cause issues if a warehouse is deleted while products still reference it. Consider using `ON DELETE CASCADE` if appropriate.